src/UIAutomation/FunctionalTests/UiaOperationAbstractionTests.cpp

```diff
@@ -2148,5 +2148,97 @@ namespace UiaOperationAbstractionTests
         {
             ElementAsVariantTest(true);
         }
+
+        TEST_METHOD(TestTryAndCatchBlockAsLocalTest)
+        {
+            TryAndCatchBlockTest(false);
+        }
+
+        TEST_METHOD(TestTryAndCatchBlockAsRemoteTest)
+        {
+            TryAndCatchBlockTest(true);
+        }
+
+        // Tests the try block and catch block implementation both locally and remote. When an exception is encountered, the execution
+        // continues till the end of instructions without returning from the exception location, also records the
+        void TryAndCatchBlockTest(const bool useRemoteOperations)
+        {
+            ModernApp app(L"Microsoft.WindowsCalculator_8wekyb3d8bbwe!App");
+            app.Activate();
+            auto calc = WaitForElementFocus(L"Display is 0");
+            UiaElement element = calc;
+
+            auto guard = InitializeUiaOperationAbstraction(useRemoteOperations);
+
+            auto scope = UiaOperationScope::StartNew();
+
+            scope.BindInput(element);
+
+            UiaInt failureCode = 0;
+            UiaArray<UiaInt> numbers;
+            numbers.Append(1);
+            numbers.Append(2);
+            numbers.Append(3);
+            scope.Try([&]()
+            {
+                auto number = numbers.GetAt(3);
+            },
+            [&](UiaFailure failure)
+            {
+                failureCode = failure.GetCurrentFailureCode();
+            });
+
+            UiaInt numberAfterTryCatch = numbers.GetAt(2);
+            scope.BindResult(failureCode);
+            scope.BindResult(numberAfterTryCatch);
+            scope.Resolve();
+
+            auto hresult = HRESULT_FROM_WIN32(failureCode);
+            bool isExpectedException = ((hresult == E_BOUNDS) || (hresult == HRESULT_FROM_WIN32(ERROR_UNHANDLED_EXCEPTION)));
```

Comment:
Huh, this is interesting that we get two failure codes. Which one is the remote one?

Reply:
E_BOUNDS is the remote one Ben

Reply:
Huh, so we get `ERROR_UNHANDLED_EXCEPTION`. Would it be difficult to fix the local error to match the remote error in this change, since they should probably be the same.

Reply:
Talked offline, for some reason it seems like `ResultFromCaughtException` isn't able to identify the exception, or something similar.